syntax = "proto3";
package spacepi.messaging;

import "spacepi/messaging/MessageID.proto";

message SubscribeRequest {
    option (MessageID) = 0x20;

    enum OperationType {
        SUBSCRIBE = 0;
        UNSUBSCRIBE = 1;
        SUBSCRIBE_ALL = 2;
        UNSUBSCRIBE_ALL = 3;
    }

    OperationType operation = 1;
    repeated uint32 messages = 2;
}
